### DISCUSSION TITLE
Browser does not Open (Chrome/Firefox)

### DISCUSSION THREAD

**Discussion by Streamline1175** [seleniumbase/SeleniumBase#3371] [General]
I have tried the different ways of getting the initial page url to be opened in either Chrome/Firefox but no luck:

```python
from seleniumbase import SB

with SB(uc=True, test=True, binary_location=bl) as sb: # Have tried a variation of args with no luck, also tried headless=False
  sb.driver.get(url)
  sb.driver.uc_open_with_reconnect(url,3)
  sb.open
```

The code runs just fine headless and pulls me what I want and gets past the Turnstile CF Captcha.

I would just like to be able to see it happen, I have just been using the save_screenshot option to track where it is getting to and what its seeing.

I am running this locally on my WSL to get it working before I push the code over to my RPI that is running my current bot code, this will be an enhanced to the codebase to fix something that was broken thanks to CF.

Here are my versions that are the latest for both chrome and firefox:

Chrome:

```bash
fire:~$ google-chrome --version
Google Chrome 131.0.6778.204 
fire:~$ chromedriver --version
ChromeDriver 131.0.6778.204 (52183f9e99a61056f9b78535f53d256f1516f2a0-refs/branch-heads/6778_155@{#7})
```

Firefox:

```bash
fire:~$ firefox --version
Mozilla Firefox 133.0.3
fire:~$ geckodriver --version
geckodriver 0.35.0 (9f0a0036bea4 2024-08-03 07:11 +0000)
```

**Comment by mdmintz** [comment]
For Linux desktops that have a GUI, add `headed=True` to see the browser.

Also note that UC Mode is for Chromium, only. (Not Firefox)

**Reply by Streamline1175** [reply]
That did the trick!
